python/lsst/qa/explorer/mkFeatures.py

```diff
@@ -0,0 +1,448 @@
+
+        These columns are added into a new index called ``featuresSG`` and more information is given in their
+        respective functions. addTruth assumes that `cleanUpCat` has been run and that it is being run on a
+        single index catalog of just the ``featuresSG`` index. This single index catalog has had the objects
+        with close neighbors (default within 1.5 arcseconds) removed in `removeNeighbors` and has been
+        narrowed down by applying a series of flag cuts in `cleanUpCat`.
+        """
+
+        cat = self.addSeeing(cat, filters)
+        cat = self.addColors(cat, filters)
+        cat = self.addMB(cat, filters)
+        cat = self.addSNFlux(cat, filters)
+        cat = self.addRaDec(cat, refCat)
+
+        cleanCat = self.cleanUpCat(cat)
+
+        medRa = np.median(cleanCat["ra"].values)
+        medDec = np.median(cleanCat["dec"].values)
+        center = afwGeom.SpherePoint(medRa, medDec, afwGeom.radians)
+        radius = (1/np.sqrt(2))*(np.max(cleanCat["ra"].values)-np.min(cleanCat["ra"].values))*afwGeom.radians
+        refs = truthRefObj.loadSkyCircle(center, radius, "MAG_BEST").refCat
+        truthCat = refs.asAstropy()
+        if len(truthCat) == 0:
+            raise RuntimeError("Truth catalogue and input data do not overlap.")
+
+        cleanCat = self.removeNeighbors(cleanCat, truthCat)
+        matchedCat = self.addTruthClass(cleanCat, truthCat)
+
+        return matchedCat
+
+    def addRaDec(self, cat, refCat):
+
+        """Add RA and Dec column from the reference catalog to the featuresSGSep index.
+
+        Parameters
+        ----------
+        cat : `pandas.core.frame.DataFrame`
+        refCat : `pandas.core.frame.DataFrame`
+            Reference RA and Declination.
+
+        Returns
+        -------
+        cat : `pandas.core.frame.DataFrame`
+            Data frame with reference RA and declination added into the featuresSGSep index.
+        """
+
+        # Reference catalogue should be in the same order as the meas catalogue (cat) but just in case use
+        # the index ids from the meas catalogue.
+        cat["featuresSGSep", "ra"] = refCat["coord_ra"][cat.index]
+        cat["featuresSGSep", "dec"] = refCat["coord_dec"][cat.index]
+
+        return cat
+
+    def cleanUpCat(self, cat):
+        """Remove objects from the dataframe that do not pass a series of flag cuts designed to clean up the
+        data. Remove the objects that are blended or saturated in any band, those that have flux <= 0 in any
+        band (psf or model flux). Also remove any NaN
+
+        Parameters
+        ----------
+        cat : `pandas.core.frame.DataFrame`
+            The multi index input data frame
+
+        Returns
+        -------
+        cleanCat : `pandas.core.frame.DataFrame`
+            A pandas data frame of only the featuresSGSep index that has been cleaned up using a series of
+            flag cuts.
+        """
+        clean = ((cat["HSC-G"]["deblend_nChild"] == 0) &
```

Comment:
This is a lot of flags to test in one line. I would suggest breaking this up into separate functions testing similar blocks. For instance testing that all the flags for a given flux and then reusing that code for the other bands.

Reply:
I've broken it up into a bunch of smaller functions to do the different sections. One for flags, one for flux cuts, one for the blendedness cut and one for the left over bits. You are right, this makes it more readable and easier to change in the future if needed.